From 194921e01f9c83834e0722e8093fd9cd94ca357d Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Fri, 30 Dec 2022 20:47:22 +0000
Subject: [PATCH 3/4] Use WebP CMake targets instead of PkgConfig

The pkgconfig module ends up with 'webp' instead of '/path/to/libwebp.lib'
xref: https://gitlab.kitware.com/cmake/cmake/-/commit/955d6245
---
 CMakeLists.txt                 |  4 ++--
 cmd/dot/CMakeLists.txt         |  8 +++-----
 plugin/webp/CMakeLists.txt     |  8 +++-----
 plugin/webp/gvloadimage_webp.c | 11 -----------
 4 files changed, 8 insertions(+), 23 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 85a458d41..d0e1d8de3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -159,6 +159,8 @@ endif()
 find_package(GD)
 find_package(GS)
 find_package(GTS)
+find_package(Threads REQUIRED)
+find_package(WEBP)
 
 if(NOT enable_ltdl STREQUAL "OFF")
   find_package(LTDL)
@@ -188,7 +190,6 @@ if(PkgConfig_FOUND)
   pkg_check_modules(LASI lasi)
   pkg_check_modules(POPPLER poppler-glib)
   pkg_check_modules(RSVG librsvg-2.0)
-  pkg_check_modules(WEBP libwebp)
   pkg_check_modules(X11 x11)
   pkg_check_modules(XRENDER xrender)
 else()
@@ -197,7 +198,6 @@ else()
   set(LASI_FOUND 0)
   set(POPPLER_FOUND 0)
   set(RSVG_FOUND 0)
-  set(WEBP_FOUND 0)
   set(X11_FOUND 0)
   set(XRENDER_FOUND 0)
 endif()
diff --git a/cmd/dot/CMakeLists.txt b/cmd/dot/CMakeLists.txt
index 96b9eb7e1..1649d2774 100644
--- a/cmd/dot/CMakeLists.txt
+++ b/cmd/dot/CMakeLists.txt
@@ -85,13 +85,11 @@ if(PkgConfig_FOUND)
   endif()
 endif()
 
+find_package(WEBP)
 if(PANGOCAIRO_FOUND)
   target_link_libraries(dot_builtins PRIVATE gvplugin_pango)
-  if(PkgConfig_FOUND)
-    pkg_check_modules(WEBP libwebp)
-    if(WEBP_FOUND)
-      target_link_libraries(dot_builtins PRIVATE gvplugin_webp)
-    endif()
+  if(WEBP_FOUND)
+    target_link_libraries(dot_builtins PRIVATE gvplugin_webp)
   endif()
 endif()
 
diff --git a/plugin/webp/CMakeLists.txt b/plugin/webp/CMakeLists.txt
index 4244e28d4..23b2f1d70 100644
--- a/plugin/webp/CMakeLists.txt
+++ b/plugin/webp/CMakeLists.txt
@@ -1,3 +1,5 @@
+find_package(WEBP)
+
 if(WEBP_FOUND)
 
   add_library(gvplugin_webp SHARED
@@ -15,13 +17,9 @@ if(WEBP_FOUND)
     ../../lib/pathplan
   )
 
-  target_include_directories(gvplugin_webp SYSTEM PRIVATE
-    ${WEBP_INCLUDE_DIRS}
-  )
-
   target_link_libraries(gvplugin_webp PRIVATE
     gvc
-    ${WEBP_LINK_LIBRARIES}
+    WebP::webp
   )
 
   if(CAIRO_FOUND)
diff --git a/plugin/webp/gvloadimage_webp.c b/plugin/webp/gvloadimage_webp.c
index 13c00268e..a42154fb2 100644
--- a/plugin/webp/gvloadimage_webp.c
+++ b/plugin/webp/gvloadimage_webp.c
@@ -23,17 +23,6 @@
 #include <cairo.h>
 #include <webp/decode.h>
 
-#ifdef _MSC_VER //*dependencies
-    #pragma comment( lib, "gvc.lib" )
-    #pragma comment( lib, "glib-2.0.lib" )
-    #pragma comment( lib, "pango-1.0.lib" )
-    #pragma comment( lib, "pangocairo-1.0.lib" )
-    #pragma comment( lib, "libcairo-2.lib" )
-    #pragma comment( lib, "gobject-2.0.lib" )
-    #pragma comment( lib, "graph.lib" )
-    #pragma comment( lib, "webp.lib" )
-#endif
-
 static const char* const kStatusMessages[] = {
     "OK", "OUT_OF_MEMORY", "INVALID_PARAM", "BITSTREAM_ERROR",
     "UNSUPPORTED_FEATURE", "SUSPENDED", "USER_ABORT", "NOT_ENOUGH_DATA"
-- 
2.45.1

